### ISSUE TITLE
Itemized deduction ceiling switch not correctly processed by TaxBrain GUI interface

### ISSUE THREAD

**Issue by hdoupe** [OpenSourcePolicyCenter/PolicyBrain#598]
I noticed that setting the switches in `Deductions subject to the cap on itemized deduction benefits` to false did not actually cause the parameter, `_ID_BenefitCap_Switch`, that was passed to `taxcalc` to be false.  I ran this [reform](http://ospc-taxes7.herokuapp.com/taxbrain/edit/1388/?start_year=2017) on a local instance of taxbrain and the value of `_ID_BenefitCap_Switch` was passed to `taxcalc` as 

``` {... u'_ID_BenefitCap_Switch': [[True, True, False, True, 1.0, 1.0, 1.0]], ...}```

However, it should have been 
``` {... u'_ID_BenefitCap_Switch': [[1.0, 1.0, 0.0, 1.0, 0.0, 1.0, 1.0]], ...}``` or 
``` {... u'_ID_BenefitCap_Switch': [[True, True, False, True, False, True, True]], ...}```

But when I upload the reform as a file, `ID_BenefitCap_Switch` is passed back as:

```{...u'_ID_BenefitCap_Switch': [[1.0, 1.0, 0.0, 1.0, 0.0, 1.0, 1.0]], ...}```

which is what I expected.

Here is a link to the test app run of the reform file: http://ospc-taxes7.herokuapp.com/taxbrain/1389/

And here are the contents of the reform file:
```
{
    "policy": {
        "_ID_BenefitCap_Switch":
            {"2017": [[1.0, 1.0, 0.0, 1.0, 0.0, 1.0, 1.0]]},
        "_ID_c":
            {"2017": [[20000, 40000, 20000, 20000, 99e9]]}
    }
}

```
So, it appears that there is an issue with the GUI interface to Taxbrain, but the file upload functionality appears to work correctly.

@brittainhard @PeterDSteinberg

**Comment by martinholmer** [comment]
@hdoupe,  Thanks for the bug report in issue #598. 
Let me be sure I understand your findings.
Are you saying that when you use the regular GUI interface to TaxBrain, the reform passed to taxcalc is correct, but when you use the file-upload interface to TaxBrain, the reform passed to TaxBrain is incorrect?  Is this right?  Or am I confused?

@brittainhard @PeterDSteinberg

**Comment by hdoupe** [comment]
@martinholmer Yes, it appears that there is an issue with the GUI interface to Taxbrain, but the file upload functionality appears to work correctly.

**Comment by martinholmer** [comment]
@hdoupe said:
> it appears that there is an issue with the GUI interface to Taxbrain, but the file upload functionality appears to work correctly.

So, I had it reversed.  You're finding that the GUI interface to TaxBrain has the reform wrong while the file-upload interface to TaxBrain has it right.  Is that correct?

**Comment by hdoupe** [comment]
@martinholmer said
> So, I had it reversed. You're finding that the GUI interface to TaxBrain has the reform wrong while the file-upload interface to TaxBrain has it right. Is that correct?

Yes, that is correct.  My apologies for the confusion.  I edited my initial comment in an effort to clear this up.

**Comment by hdoupe** [comment]
Closed via #641
